spm-monitor-generic/src/main/java/com/sematext/spm/client/postgresql/PostgreSQLVersionCheck.java

```diff
@@ -0,0 +1,82 @@
+package com.sematext.spm.client.postgresql;
+
+import com.sematext.spm.client.Log;
+import com.sematext.spm.client.LogFactory;
+import com.sematext.spm.client.db.DbConnectionManager;
+import com.sematext.spm.client.db.DbStatsExtractorConfig;
+import com.sematext.spm.client.observation.BaseVersionConditionCheck;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PostgreSQLVersionCheck extends BaseVersionConditionCheck {
+    private static final Log LOG = LogFactory.getLog(PostgreSQLVersionCheck.class);
+
+    @Override
+    protected String readVersion() {
+        DbStatsExtractorConfig config = (DbStatsExtractorConfig) getExtractorConfig();
+
+        String dbDriverClass = config.getDbDriverClass();
+        String dbUrl = config.getDbUrl();
+        String dbUser = config.getDbUser();
+        String dbPassword = config.getDbPassword();
+        String dbAdditionalConnectionParams = config.getDbAdditionalConnectionParams();
+
+        DbConnectionManager dbConnectionManager = DbConnectionManager.getConnectionManager(dbUrl);
+
+
+        try {
+            if (dbConnectionManager == null) {
+                dbConnectionManager = new DbConnectionManager(dbUrl, dbDriverClass, dbUser, dbPassword, dbAdditionalConnectionParams);
+            }
+
+            Connection conn = dbConnectionManager.getConnection();
+            if (conn == null) {
+                LOG.error("DB connection not available, skipping query");
+
+                return null;
+            }
+
+            Statement stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery("SHOW SERVER_VERSION;");
+
+            if (rs.next()) {
+                String version = rs.getString(1).trim();
+
+                int[] versions;
+                try {
+                    String[] versionParts = version.split(" ")[0].split(".");
+                    versions = new int[versionParts.length];
+                    for(int i = 0; i < versionParts.length; i++) {
+                        versions[i] = Integer.parseInt(versionParts[i]);
+                    }
+
+                    return version;
+                } catch (Exception e) {
+                    // Postgres might be in development, with format \d+[beta|rc]\d+
+                    Pattern pattern = Pattern.compile("(\\d+)([a-zA-Z]+)(\\d+)");
+                    Matcher m = pattern.matcher(version);
+                    if (m.matches() && m.groupCount() == 3) {
+                        versions = new int[3];
+                        versions[0] = Integer.parseInt(m.group(1));
+                        versions[1] = -1;
+                        versions[2] = Integer.parseInt(m.group(3));
+                    } else {
+                        return null;
+                    }
+
+                    return versions[0] + ".-1." + versions[2]; //TODO check if support 2.-1.3 version (which is < 2.0)
```

Comment:
I don't understand what this TODO is saying...

Could we implement now whatever this TODO is saying instead of leaving the TODO?

Reply:
Postgresql beta/rc version have format "11 Beta 3" i convert it to 11.-1.3, normally this version is before 11.0 (that why i chose minor version is -1)
Im not sure if my format 11.-1.3 matching 11.* in yml require section.

Reply:
I don't think we need/should be providing versions-specific integration configs for beta versions (WDYT @bsmid?).  Also, PG 11 (11.1 actually) seems to already have been released, so not sure why beta/rc is even being mentioned here.  c.f. https://en.wikipedia.org/wiki/PostgreSQL#Release_history
https://www.postgresql.org/about/news/1905/

Reply:
I think we should convert "11 Beta 3" to just "11" and use just "11" in our configs.